api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/CommonQueries.java

```diff
@@ -139,8 +143,38 @@ public static String getPatientsOnTbTreatmentQuery(
             + "  AND e.voided=0 "
             + "  AND o.voided=0 "
             + "  AND p.voided=0 "
-            + "  AND e.location_id =   :location ; "
-            + "";
+            + "  AND e.location_id =   :location  "
+            + " UNION "
+            + " SELECT p.patient_id "
+            + "            FROM patient p "
+            + "                INNER JOIN encounter e  "
+            + "                    ON p.patient_id = e.patient_id "
+            + "                INNER JOIN obs o  "
+            + "                    ON e.encounter_id = o.encounter_id "
+            + "                INNER JOIN ("
+            + "                    SELECT p.patient_id, max(o.obs_datetime) obsdatetime"
+            + "                    FROM patient p "
+            + "                        INNER JOIN encounter e "
+            + "                            ON p.patient_id = e.patient_id "
+            + "                        INNER JOIN obs o "
+            + "                        ON e.encounter_id = o.encounter_id "
+            + "                    WHERE o.concept_id = ${pulmonaryTBConceptId} "
+            + "                        AND e.location_id =   :location  "
+            + "                        AND e.encounter_type = ${masterCardEncounterType} "
+            + "                        AND o.obs_datetime BETWEEN date_sub(:endDate, INTERVAL 6 MONTH) AND :endDate "
+            + "                        AND p.voided = 0 "
+            + "                        and e.voided = 0 "
+            + "                        and o.voided = 0 "
+            + "                    GROUP BY p.patient_id"
```

Comment:
Isnt this the case where we you have to first pick the most recent then see if that has this concept on it cause this way it will filter all that has this pulmonary concept??
@cfaife please confirm!

Reply:
ok I will confirm